### DISCUSSION TITLE
Request to Automatically Activate Routes from NMEA Messages

### DISCUSSION THREAD

**Discussion by wbijen** [antipole2/JavaScript_pi#97] [Q&A]
First of all, wow! What an impressive library you've created. Very cool!

I've been using the script "Build_routes_from_NMEA.js" for generating routes from NMEA messages, and it's working wonderfully for my use case. However, I would like to enhance its functionality by having the option to automatically activate the routes as soon as they are created. Is this currently possible, or does this feature need to be implemented?

Could you please provide some guidance or consider adding this feature?

Thank you!

**Comment by antipole2** [comment]
@wbijen Glad you have found that script useful. It was written some long while ago for some university researchers working on collision avoidance strategies.   Perhaps you would like to share a bit about how you are using it?  Like where your NMEA data is coming from.
 
All plugins rely on the APIs to interact with OpenCPN.  I do not believe there is any API to activate a route, so there is nothing that can be done at present.
I vaguely recall making a request for adding such an API some long time ago but that has fallen by the wayside.  There has to be a good case for such an addition.  Hence, my interest in your application.

**Comment by wbijen** [comment]
Thank you for your prompt response and for providing some background on the development of the script. Interesting to hear about its origins.

I am using the script in a somewhat different context. I plan my routes using the C-MAP app on my phone, which allows me to export them in GPX format. I have developed a small mobile app that converts these GPX routes into NMEA messages, enabling them to be sent to OpenCPN. My setup involves running OpenCPN on a Raspberry Pi as a board computer in a headless configuration, which means I rely heavily on automation (or vnc viewer).

My phone should act as the remote. And I aim for the Raspberry Pi to handle the routing. Once a route is activated in OpenCPN, the autopilot software pypilot would then take over. It is also using the [autopilot route plugin](https://github.com/seandepagnier/autopilot_route_pi)

This setup would enable a nice transition from route planning to navigation without manual intervention, which is critical for maintaining the headless operation setup I am aiming for. Understanding that there currently isn't an API for activating a route directly, I wonder if revisiting your previous request for such an API might now have a strong case given the automation and integration capabilities it could enable in setups like mine.

I appreciate your interest in my application, maybe there are better ways to accomplish my goals but I'm not aware of any.

Thank you!

**Comment by antipole2** [comment]
I have submitted a request for the provision of an API to allow this.  [See here](https://github.com/OpenCPN/OpenCPN/issues/3845).

**Comment by antipole2** [comment]
It looks like the necessary API will be in OpenCPN v5.14
It would require an update to the JavaScript plugin to make it available to scripts.
